stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro

```diff
@@ -44,24 +49,44 @@
 ;
 ; :history:
 ;    30-Jun-2022 - ECMD (Graz), initial release
+;    15-Jul-2022 - Andrea Francesco Battaglia (FHNW)
+;                  Added a few functionalities:
+;                       -> automatic recognition of SPEC and CPD files: /is_pixel_data is no more needed
+;                       -> option for manually define time_shift. Default: assume solar center
+;                       -> rate keyword added
+;                          
 ;
 ;-
 function stx_science_data_lightcurve, fits_path, energy_ranges = edges_in,  time_min = time_min,  $
-  fits_path_bk =  fits_path_bk, is_pixel_data = is_pixel_data, plot_obj = plot_obj
+  fits_path_bk =  fits_path_bk, plot_obj = plot_obj, time_shift = time_shift, rate = rate, shift_duration = shift_duration
 
   default, time_min, 20
   default, edges_in, [[4.,10.],[10,15],[15,25]]
-
+  default, spex_units, 'flux'
+  
+  ; If /rate is set, return the rate units
+  if keyword_set(rate) then spex_units = 'rate'
+  
   ;for the light curve the standard default corrections are applied
-  stx_get_header_corrections, fits_path, distance = distance, time_shift = time_shift
+  ; If the user manually defines time_shift, then use that
+  if keyword_set(time_shift) then begin
+    stx_get_header_corrections, fits_path, distance = distance
+  endif else begin
+    stx_get_header_corrections, fits_path, distance = distance, time_shift = time_shift
+  endelse
 
   edge_products, edges_in, edges_2 = energy_ranges
-
-
-  if keyword_set(is_pixel_data) then begin
-    stx_convert_pixel_data, fits_path_data = fits_path, fits_path_bk =  fits_path_bk, distance = distance, time_shift = time_shift, ospex_obj = ospex_obj, _extra= _extra
-  endif else begin
+  
+  ; Get the original filename
+  !null = mrdfits(fits_path, 0, primary_header)
+  orig_filename = sxpar(primary_header, 'FILENAME')
+
+  if strpos(orig_filename, 'cpd') gt -1 then begin
+    stx_convert_pixel_data, fits_path_data = fits_path, fits_path_bk =  fits_path_bk, distance = distance, time_shift = time_shift, ospex_obj = ospex_obj, shift_duration = shift_duration, _extra= _extra
+  endif else if strpos(orig_filename, 'spec') gt -1 then begin
     stx_convert_spectrogram, fits_path_data = fits_path, fits_path_bk =  fits_path_bk, distance = distance, time_shift = time_shift, ospex_obj = ospex_obj, _extra= _extra
+  endif else begin
+    message, 'ERROR: the FILENAME field in the primary header should contain either cpd or spec'
   endelse
```

Comment:
I think this won't work with the L1A files which have a different naming scheme? In the short term it would be good to be able to use them too.

Reply:
Ok, I will create a new pull request allowing the alpha version files too.